### PR TITLE
Let #nullable directive enable/disable all nullable-related warnings

### LINKED ISSUE #32742: Should #nullable and #pragma nullable have effect on all warnings related to nullable feature rather than only on those reported by flow analysis?
Right now ```#nullable``` and ```#pragma nullable``` have effect only on warnings that are reported by flow analysis. Other nullable related warnings, for example, for various mismatches in declarations are enabled by default. Should all those warnings be disabled by default and obey ```#nullable``` and ```#pragma nullable``` settings?

### PR DESCRIPTION
**Pull request by RikkiGibson**
Resolves #32742

Would like to know what people think about:
1. the decision to control all nullable-related warnings via `#nullable enable warnings` (except using the annotation outside of nullable annotations context warning)
2. this solution to ensure via tests/automated checks that any new nullable-related warnings are either manually added to the set or marked as non-nullable-related

### CODE REVIEW

**Comment by jcouv** [comment]
>         public void Directive_Annotations_NonFlowAnalysisWarning_01()

Can you add a variant with `#nullable enable`?

---
Refers to: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs:5591 in 55f7182. [](commit_id = 55f7182f56b8943acb2cf816ffab213e2346a170, deletion_comment = False)

**Comment by jcouv** [comment]
>             );

technically, we should find a test with each warning ID that you added to the enum and verify the behavior. That said, I'd be fine to only do that for 2 or 3 of them and filling an issue for the remaining ones.
I'll let the team chime in on how much validation we want

---
Refers to: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs:70705 in 55f7182. [](commit_id = 55f7182f56b8943acb2cf816ffab213e2346a170, deletion_comment = False)

File: src/Compilers/CSharp/Portable/Errors/ErrorFacts.cs [hunk 1]
```diff
@@ -15,16 +15,60 @@ internal static partial class ErrorFacts
...
         private static readonly Lazy<ImmutableDictionary<ErrorCode, string>> s_helpLinksMap = new Lazy<ImmutableDictionary<ErrorCode, string>>(CreateHelpLinks);
         private static readonly Lazy<ImmutableDictionary<ErrorCode, string>> s_categoriesMap = new Lazy<ImmutableDictionary<ErrorCode, string>>(CreateCategoriesMap);
-        public static readonly ImmutableHashSet<string> NullableFlowAnalysisWarnings;
+        public static readonly ImmutableHashSet<string> NullableWarnings;
 
         static ErrorFacts()
         {
             ImmutableHashSet<string>.Builder builder = ImmutableHashSet.CreateBuilder<string>();
+
+            builder.Add(getId(ErrorCode.WRN_ConvertingNullableToNonNullable));
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Errors/ErrorFacts.cs]: I realized this is full of dupes, will fix (probably by using a minimal diff and adding the new ones to the end)

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler Let's review this asap and merge before noon deadline if possible. Thanks

File: src/Compilers/CSharp/Test/Syntax/Diagnostics/DiagnosticTest.cs [hunk 2]
```diff
@@ -323,6 +323,38 @@ public void WarningLevel_2()
...
+                    continue;
+                }
+
+                if (ErrorFacts.NullableWarnings.Contains(MessageProvider.Instance.GetIdForErrorCode((int)error)))
+                {
+                    continue;
+                }
+
+                // Nullable-unrelated warnings in the C# 8 range should be added to this array.
+                var nullableUnrelatedWarnings = new[]
```
Comment (agocke) [inline src/Compilers/CSharp/Test/Syntax/Diagnostics/DiagnosticTest.cs]: We won't be adding new warnings in C# 8 so this validates the existing code but doesn't really protect against the future, where we add warnings in future language versions.

Comment (RikkiGibson) [reply src/Compilers/CSharp/Test/Syntax/Diagnostics/DiagnosticTest.cs]: Yeah, let's consider how we want to handle this moving forward (it seems like we wouldn't expect most of the warnings in C# 9 to be nullable-related?)

**Pull request merged**